winsup/cygwin/uinfo.cc

```diff
@@ -2438,6 +2438,45 @@ pwdgrp::fetch_account_from_windows (fetch_user_arg_t &arg, cyg_ldap *pldap)
     p = wcpcpy (wcpcpy (p, dom), cygheap->pg.nss_separator ());
   wcpcpy (p, name);
 
+  if (!home)
+    {
+      char *env = NULL;
+      env = getenv("HOME");
+      if (!env)
+        {
+          char *drive = getenv("HOMEDRIVE"), *path = getenv("HOMEPATH");
+          if (drive && path)
+            {
+              int drive_len = strlen(drive), path_len = strlen(path);
+              home = (char *)malloc(drive_len + path_len + 1);
+              memcpy(home, drive, drive_len);
+              memcpy(home + drive_len, path, path_len);
+              home[drive_len + path_len] = '\0';
```

Comment:
Instead of the three lines above, why not these two:

strcpy(home, drive);
strcpy(home + drive_len, path);

That would also allow us to get rid of the path_len variable.